### PR TITLE
vpc-nat-gw: configure routes for subnets by pod annotations

### PR DESCRIPTION
**Pull request by zhangzujian**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Features
- Bug fixes
- Docs
- Tests

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

File: pkg/controller/subnet.go [hunk 1]
```diff
@@ -3241,3 +3241,37 @@ func (c *Controller) deletePolicyRouteForU2ONoLoadBalancer(subnet *kubeovnv1.Sub
...
+	}
+	var subnet *kubeovnv1.Subnet
+	for _, s := range subnets {
+		if s.Spec.Provider == provder {
+			subnet = s.DeepCopy()
+			break
+		}
+	}
+	if subnet == nil {
+		err = fmt.Errorf("failed to get subnet for net-attach-def %s/%s", ns, name)
```
Comment (bobz965) [inline pkg/controller/subnet.go:3272]: log near err

File: pkg/controller/vpc_nat_gateway.go [hunk 2]
```diff
@@ -367,15 +370,15 @@ func (c *Controller) handleInitVpcNatGw(key string) error {
 
 	if pod.Status.Phase != corev1.PodRunning {
 		time.Sleep(10 * time.Second)
-		return fmt.Errorf("failed to init vpc nat gateway, pod is not ready")
+		return fmt.Errorf("failed to init vpc nat gateway %s, pod is not ready", key)
```
Comment (bobz965) [inline pkg/controller/vpc_nat_gateway.go:373]: log near err

File: pkg/controller/vpc_nat_gw_eip.go [hunk 3]
```diff
@@ -299,15 +314,28 @@ func (c *Controller) handleUpdateIptablesEip(key string) error {
...
 
-	externalNetwork := util.GetExternalNetwork(cachedEip.Spec.ExternalSubnet)
-	// should delete
+	subnets, err := c.subnetsLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("failed to list subnets: %v", err)
+		return err
+	}
+
+	nadName := util.GetExternalNetwork(cachedEip.Spec.ExternalSubnet)
```
Comment (bobz965) [inline pkg/controller/vpc_nat_gw_eip.go:323]: It is better to put `subnets, err := c.subnetsLister.List(labels.Everything())` after `c.subnetsLister.List(`

Comment (zhangzujian) [reply pkg/controller/vpc_nat_gw_eip.go:323]: What's the difference?

Comment (bobz965) [reply pkg/controller/vpc_nat_gw_eip.go:323]: nadName := util.GetExternalNetwork 更容易出错，感觉应该把更容易出错的放前面

Comment (bobz965) [reply pkg/controller/vpc_nat_gw_eip.go:323]: 噢  我仔细看了下 util.GetExternalNetwork(，发现里面不会出错。我理解错了。

**Review by bobz965 — APPROVED** [review]

**Pull request merged**